inst/pages/exercises.qmd

```diff
@@ -1741,6 +1741,188 @@ pheatmap(
+tse <- GlobalPatterns
+
+# Calculate the Shannon diversity index
+diversity <- estimateDiversity(tse, assay.type = "counts", index = "shannon")
+
+# Extract the Shannon diversity index and bmi_group
+shannon_index <- diversity$shannon
+bmi_group <- colData(tse)$SampleType  # Using SampleType as a proxy for BMI_Group
+
+# Create the data frame
+data <- data.frame(
+  Shannon_Index = shannon_index,
+  BMI_Group = bmi_group
+)
+
+# Perform the regression analysis
+model <- lm(Shannon_Index ~ BMI_Group, data = data)
+
+# Create the plot with the regression line
+p <- ggplot(data, aes(x = BMI_Group, y = Shannon_Index)) +
+  geom_point(color = 'blue') +
+  geom_smooth(method = 'lm', color = 'red') +
+  labs(title = 'Regression Analysis of Shannon Index vs. BMI Group',
+       x = 'BMI Group', y = 'Shannon Index') +
+  theme_minimal()
+
+# Print the plot
+print(p)
+
+# Print the model summary
+summary(model)
+```
+What do we interpret from the above regression analysis?
+
+1. The coefficients of the regression model indicate the relationship 
+between the BMI groups and the Shannon diversity index. If the coefficient
+for a particular BMI group is significantly different from zero, it
+suggests that this group has a different Shannon diversity index compared
+to the reference group.
+2. The p-value associated with each coefficient tells us whether the effect
+of that BMI group on the Shannon diversity index is statistically
+significant. A common threshold for significance is 0.05.
+3. R-squared: This value indicates the proportion of the variance in the
+Shannon diversity index that is explained by the BMI groups. A higher R-squared
+value indicates a better fit of the model.
+
+Now, let's check some linear modeling assumptions :
+
+```{r}
+#| label: sub-checking-modeling-assumptions
+
+# 1. Linearity
+plot(model, which = 1)
+
+# 2. Normality of residuals
+plot(model, which = 2)
+
+# 3. Homoscedasticity
+plot(model, which = 3)
+
+# 4. Independence of residuals (Durbin-Watson test)
+dwtest(model)
+```
+Here, 
+1. Linearity plot(Plot 1) shows the residuals vs. fitted values. The residuals
+should be randomly scattered around zero, indicating that the linearity
+assumption is met. 
+2. Normality/QQ plot(Plot 2) of the residuals should show that the residuals
+are approximately normally distributed. Deviations from the line suggest 
+deviations from normality.
+3. Homoscedasticity (Plot 3): The residuals vs. fitted values plot should
+show constant variance of the residuals (homoscedasticity). If there is a
+pattern (e.g., funnel shape), it indicates heteroscedasticity.
+4. Durbin-Watson Test checks for autocorrelation in the residuals. A value
+close to 2 indicates no autocorrelation. Values significantly different 
+from 2 suggest autocorrelation.
+
+Further, treat sex as another possible covariate :
+
+```{r}
+#| label: sub-trying-possible-covariates
+
+sex <- colData(tse)$SampleType
+
+# Add sex to the data frame
+data$Sex <- sex
+
```

Comment:
again, no need to create new variables as you can directly run models with original data